src/herder/TxSetFrame.cpp

```diff
@@ -790,57 +823,58 @@ TxSetFrame::addTxsFromXdr(Hash const& networkID,
 }
 
 void
-TxSetFrame::surgePricingFilter(uint32_t opsLeft)
+TxSetFrame::applySurgePricing(Application& app)
 {
     ZoneScoped;
-    auto curSizeOps = sizeOp();
-    if (curSizeOps <= opsLeft)
+
+    if (mTxs.empty())
     {
+        mFeesComputed = true;
         return;
     }
-    CLOG_WARNING(Herder, "surge pricing in effect! {} > {}", curSizeOps,
-                 opsLeft);
 
-    auto actTxQueueMap = TxSetUtils::buildAccountTxQueues(mTxs);
+    size_t maxOps = app.getLedgerManager().getLastMaxTxSetSizeOps();
```

Comment:
too many changes in the same commit: this is mixing the refactor (to use surge pricing utils) and adding lane support/dex